spec/generators/rails-template.rb

```diff
@@ -114,28 +114,72 @@ def wood_type
     end
   RUBY
 
-  file "app/models/wizard.rb", <<~RUBY
-    class Wizard < ApplicationRecord
-      validates :name, length: { minimum: 5 }, presence: true
-
-      enum house: {
-        Gryffindor: 0,
-        Hufflepuff: 1,
-        Ravenclaw: 2,
-        Slytherin: 3,
-      }
-
-      enum broom: {
-        nimbus: 'nimbus',
-        firebolt: 'firebolt',
-      }
-
-      has_one :wand
-      has_many :spell_books
-
-      scope :recent, -> { where('created_at > ?', 1.month.ago) }
-    end
-  RUBY
+  if ENV["RAILS_VERSION"] == "4.2"
+    file "app/models/wizard.rb", <<~RUBY
+      class Wizard < ApplicationRecord
+        validates :name, length: { minimum: 5 }, presence: true
+
+        enum house: {
+          Gryffindor: 0,
+          Hufflepuff: 1,
+          Ravenclaw: 2,
+          Slytherin: 3,
+        }
+
+        enum broom: {
+          nimbus: 'nimbus',
+          firebolt: 'firebolt',
+        }
+
+        has_one :wand
+        has_many :spell_books
+
+        scope :recent, -> { where('created_at > ?', 1.month.ago) }
+      end
+    RUBY
+  else
+    file "app/models/wizard.rb", <<~RUBY
+      class Wizard < ApplicationRecord
+        validates :name, length: { minimum: 5 }, presence: true
+
+        enum house: {
+          Gryffindor: 0,
+          Hufflepuff: 1,
+          Ravenclaw: 2,
+          Slytherin: 3,
+        }
+
+        enum broom: {
+          nimbus: 'nimbus',
+          firebolt: 'firebolt',
+        }, _prefix: true
+
+        enum quidditch_position: {
+          keeper: 0,
+          seeker: 1,
+          beater: 2,
+          chaser: 3,
+        }, _prefix: :quidditch
+
+        enum hair_color: {
+          brown: 0,
+          black: 1,
+          blonde: 2,
+        }, _suffix: :hair
+
+        enum eye_color: {
```

Comment:
sure! the point of this one was from some earlier testing, because the old system would have declared 2 `brown?` methods (one for hair, one for eyes) and I was trying to figure out a better way to get the methods. Now, tho, I'll add a prefix/suffix case